fastfood/food.py

```diff
@@ -92,15 +81,22 @@ def update_cookbook(cookbook, templatepack, stencilset_name, **options):
     files = {
         # files.keys() are template paths, files.values() are target paths
         # {path to template: rendered target path, ... }
-        os.path.join(stencilset.path, tpl): os.path.join(cookbook.path, tgt)
+        os.path.join(stencilset.path, tpl): os.path.join(cookbook_path, tgt)
         for tgt, tpl in stencil['files'].iteritems()
     }
 
     template_map = {
+        'cookbook': { "name": cookbook_name },
         'options': stencil['options'],
-        'cookbook': cookbook.metadata.to_dict().copy(),
     }
 
+    # Cookbooks may not yet have metadata, so we pass an empty dict if so
+    try:
+        cookbook = book.CookBook(cookbook_path)
+        template_map['cookbook'] = cookbook.metadata.to_dict().copy()
+    except:
```

Comment:
Lets catch a more specific Exception here.

In general you want to avoid writing `except:`

> Because except: catches all exceptions, including SystemExit, KeyboardInterrupt, and GeneratorExit (which is not an error and should not normally be caught by user code), using a bare except: is almost never a good idea. In situations where you need to catch all “normal” errors, such as in a framework that runs callbacks, you can catch the base class for all normal exceptions, Exception. Unfortunately in Python 2.x it is possible for third-party code to raise exceptions that do not inherit from Exception, so in Python 2.x there are some cases where you may have to use a bare except: and manually re-raise the exceptions you don’t want to catch.

https://docs.python.org/2.7/howto/doanddont.html#except